### DISCUSSION TITLE
Random overrides

### DISCUSSION THREAD

**Discussion by maxfrei750** [facebookresearch/hydra#2446] [Q&A]
The extended override syntax offers a range of options, yet I'm missing a random choice option. Something like
```
randomchoice(range(10), 5)
```
which would give me 5 entries from the range [0, 9]. Is there a workaround?

Thanks in advance.

**Comment by Jasha10** [comment]
Hydra's sweeper plugins have some random choice functionality. See e.g. [Optuna Sweeper range override](https://hydra.cc/docs/1.2/plugins/optuna_sweeper/#range-override).

**Reply by maxfrei750** [reply]
Thanks for point that out! However, it would be nice to use the default sweeper for such a basic task.

**Comment by omry** [comment]
There is a [shuffle](https://hydra.cc/docs/advanced/override_grammar/extended/#shuffle) command which gets close but does not offer what you are after.
![image](https://user-images.githubusercontent.com/376455/198331408-44df3280-df13-4324-9d51-fc7c7ebc6ecf.png)

Maybe an additional command to select a sublist would be sufficient here.
```
select(shuffle([1,2,3], 0, 2)
```

Maybe a better solution would be to add the ability to produce a subset of the output in shuffle.
```
shuffle(list=[1,2,3], end=2)
shuffle(list=[1,2,3], start=1, end=2)
```

**Reply by maxfrei750** [reply]
I think that both solutions would work just fine. IMHO, `select` is more explicit

Edit: Thanks btw for the awesome work you're doing.
